package/kernel/linux/modules/lib.mk

```diff
@@ -7,6 +7,79 @@
 
 LIB_MENU:=Libraries
 
+define KernelPackage/lib-crypto-blake2s
+  SUBMENU:=$(LIB_MENU)
+  TITLE:=BLAKE2s hash function library
+  KCONFIG:=CONFIG_CRYPTO_LIB_BLAKE2S
+  FILES_PLATFORM:=$(LINUX_DIR)/lib/crypto/libblake2s-generic.ko
+  FILES:=$(LINUX_DIR)/lib/crypto/libblake2s.ko $(FILES_PLATFORM)
+endef
+
+define KernelPackage/lib-crypto-blake2s/x86/64
+  KCONFIG+=CONFIG_CRYPTO_BLAKE2S_X86
+  FILES_PLATFORM:=$(LINUX_DIR)/arch/x86/crypto/blake2s-x86_64.ko
+endef
+
+$(eval $(call KernelPackage,lib-crypto-blake2s))
+
+
+define KernelPackage/lib-crypto-chacha20poly1305
+  SUBMENU:=$(LIB_MENU)
+  TITLE:=ChaCha20-Poly1305 AEAD support (8-byte nonce library version)
+  KCONFIG:=CONFIG_CRYPTO_LIB_CHACHA20POLY1305
+  FILES_PLATFORM:= \
+	$(LINUX_DIR)/lib/crypto/libpoly1305.ko \
+	$(LINUX_DIR)/lib/crypto/libchacha.ko
+  FILES:=$(LINUX_DIR)/lib/crypto/libchacha20poly1305.ko $(FILES_PLATFORM)
+endef
+
+define KernelPackage/lib-crypto-chacha20poly1305/x86/64
+  KCONFIG+= \
+	CONFIG_CRYPTO_CHACHA20_X86_64 \
+	CONFIG_CRYPTO_POLY1305_X86_64
+  FILES_PLATFORM:= \
+	$(LINUX_DIR)/arch/x86/crypto/poly1305-x86_64.ko \
+	$(LINUX_DIR)/arch/x86/crypto/chacha-x86_64.ko
```

Comment:
Should be split into one for chacha and one for poly.

libchacha20poly1305 is one library. It uses libchacha20 and libpoly1305.

Reply:
I'll try and figure out a better way to express this :)

Reply:
I think we want:
kmod-lib-crypto-chacha20poly1305
              /                                               \
 kmod-lib-crypto-chacha20             kmod-lib-crypto-poly1305

Reply:
Right.